packages/core-plugin-api/src/plugin/types.ts

```diff
@@ -62,6 +70,7 @@ export type BackstagePlugin<
   provide<T>(extension: Extension<T>): T;
   routes: Routes;
   externalRoutes: ExternalRoutes;
+  __experimentalReconfigure(options: PluginInputOptions): void;
```

Comment:
This needs to use a type parameter to make the proper input options type for each plugin available. We could even remove the reconfigure method if there is no configure method by having the type parameter be `never` in that case

Reply:
@Rugvip Made it strict to a plugin definition type.